amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/QueueServiceImpl.java

```diff
@@ -0,0 +1,295 @@
+            } else {
+                if (null == amqpQueue) {
+                    channel.closeChannel(ErrorCodes.NOT_FOUND, "No such queue: " + finalQueue);
+                } else {
+                    channel.checkExclusiveQueue(amqpQueue);
+                    channel.setDefaultQueue(amqpQueue);
+
+                    MethodRegistry methodRegistry = connection.getMethodRegistry();
+                    QueueDeclareOkBody responseBody = methodRegistry.createQueueDeclareOkBody(finalQueue, 0, 0);
+                    connection.writeFrame(responseBody.generateFrame(channelId));
+                }
+            }
+        });
+    }
+
+    @Override
+    public void queueDelete(AmqpChannel channel, AMQShortString queue, boolean ifUnused,
+                            boolean ifEmpty, boolean nowait) {
+        int channelId = channel.getChannelId();
+        AmqpConnection connection = channel.getConnection();
+        if (log.isDebugEnabled()) {
+            log.debug("RECV[{}] QueueDelete[ queue: {}, ifUnused:{}, ifEmpty:{}, nowait:{} ]", channelId, queue,
+                    ifUnused, ifEmpty, nowait);
+        }
+        if ((queue == null) || (queue.length() == 0)) {
+            //get the default queue on the channel:
+            AmqpQueue amqpQueue = channel.getDefaultQueue();
+            delete(channel, amqpQueue);
+        } else {
+            CompletableFuture<AmqpQueue> amqpQueueCompletableFuture =
+                    queueContainer.asyncGetQueue(connection.getNamespaceName(), queue.toString(), false);
+            amqpQueueCompletableFuture.whenComplete((amqpQueue, throwable) -> {
+                if (throwable != null) {
+                    log.error("Get Topic error:{}", throwable.getMessage());
+                    channel.closeChannel(INTERNAL_ERROR, "Get Topic error: " + throwable.getMessage());
+                } else {
+                    if (null == amqpQueue) {
+                        channel.closeChannel(ErrorCodes.NOT_FOUND, "No such queue: " + queue.toString());
+                    } else {
+                        delete(channel, amqpQueue);
+                    }
+                }
+            });
+        }
+    }
+
+    @Override
+    public void queueBind(AmqpChannel channel, AMQShortString queue, AMQShortString exchange, AMQShortString bindingKey,
+                          boolean nowait, FieldTable argumentsTable) {
+        int channelId = channel.getChannelId();
+        AmqpConnection connection = channel.getConnection();
+        if (log.isDebugEnabled()) {
+            log.debug("RECV[{}] QueueBind[ queue: {}, exchange: {}, bindingKey:{}, nowait:{}, arguments:{} ]",
+                    channelId, queue, exchange, bindingKey, nowait, argumentsTable);
+        }
+        Map<String, Object> arguments = FieldTable.convertToMap(argumentsTable);
+        if (queue == null || StringUtils.isEmpty(queue.toString())) {
```

Comment:
Do we really want to "guess" which queue to use?  I would think that if the `queue` is null, you should throw an `IllegalArgumentException`

Reply:
If the queue is null，we should use the default queue, which was declare in the last of this channel.